Strategy11/formidable-forms
# In-plugin summary emails

**The Overview:** https://app.clickup.com/9006050445/v/dc/8ccuv4d-11611

The design for monthly and yearly email and the new settings: https://www.figma.com/file/jI9WfZElTkZIz8w9gAlVTu/Monthly%2C-Yearly-Review-Email?node-id=90%3A507&t=2dz6EarZGT3SB05Q-1 

**The technical pitch:** https://app.clickup.com/9006050445/v/dc/8ccuv4d-11571/8ccuv4d-5351

**Development tasks list:**
- [x] Create new settings
- [x] Add cron event
- [x] Add code to check when should send emails
- [x] Add base code for sending emails
- [x] Collect monthly data
- [x] Build monthly email content
- [x] Build yearly email content
- [x] Build license expired email content

**Test cases:**
- [ ] There will be a new checkbox in the Misc section of Global settings, it should be enabled by default, and the Recipients setting displays. We will support `[admin_email]` in the Recipients setting.
- [ ] Make sure that no monthly or yearly email is sent in the first 15 days after updating.
- [ ] After 15 days, if it's less than 45 days to the renewal date, or the oldest form was created at least 1 year ago, a yearly email will be sent. Otherwise, a monthly email will be sent.
- [ ] Make sure that the time between 2 monthly or 1 monthly and 1 yearly email is at least 30 days.
- [ ] If the license is expired, a license expired email should be sent. And it's sent just once.
- [ ] Make sure emails are sent to the correct recipients (set in the new settings).
- [ ] To send email immediately, run the following code:

```
add_action( 'init', function() {
    if ( method_exists( 'FrmEmailSummaryHelper', 'send_monthly') ) {
        FrmEmailSummaryHelper::send_monthly(); // send monthly email for the last 30 days.
        FrmEmailSummaryHelper::send_yearly(); // send yearly email for the last 1 year.
    }
} );
```

Sent date tracking is in the option `frm_summary_emails_options`.

Use `add_filter( 'frm_html_summary_emails', '__return_false' );` to test the plain text emails.

**Note:** License expired email is moved to Formidable Pro. To test it, get the code in https://github.com/Strategy11/formidable-pro/pull/4594 then run `FrmProEmailLicenseExpiredController::send()`.

**QA site:**
I set up the QA site and added 3 snippets to send the summary emails. They are [here](https://qa.formidableforms.com/truong/wp-admin/admin.php?page=snippets), just run one of them once to test. Before running, make sure that you added your email address in `Formidable > Global settings > Miscellaneous > Recipients`.

Currently, I can't make the out-of-date plugins section included in the email. Look like it's related to multisite. I'm trying to make it work.

@stephywells You should review the content of those emails. Some URLs I'm not sure they are correct.
@tuguirazvan Please take a look at those emails and give me the design feedback.
